### ISSUE TITLE
UserWarning: None of the inputs have requires_grad=True. Gradients will be None

### ISSUE THREAD

**Issue by gumona12** [2U1/Llama3.2-Vision-Finetune#27]
Hi!

Even though I set 

```
    --tune_img_projector False \
    --freeze_vision_tower True \
    --freeze_llm False \
```

There is warning about `None of the inputs have requires_grad=True. Gradients will be None`

Do you know what is the problem?

Thank you!

**Comment by 2U1** [comment]
@gumona12 Okay I found the reason.
Some the gradietns are set to False when you freeze the vision tower using gradient checkpointing.
It doesn't matter when training, so it just fine.
